.github/workflows/test.yml

```diff
@@ -0,0 +1,21 @@
+name: JavaScript CI
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: npm ci
+      - run: npm run build
```

Comment:
This one seems unneeded because `npm ci` already run it.
```suggestion
```

You can also remove all other scripts than `build` in the `package.json`